src/ws/metrics.rs

```diff
@@ -1,10 +1,43 @@
+use std::sync::Arc;
+
+use tokio::sync::Mutex;
 use warp::{http::header::HeaderValue, reject::Reject, Rejection, Reply};
 
-use lazy_static::lazy_static;
 use prometheus::{self, Encoder, IntCounterVec, Opts, Registry};
 
 use super::service::mapper::error::WsServiceError;
 
+#[derive(Clone)]
+pub struct Metrics {
+    pub procedure_call_collector: IntCounterVec,
+    pub registry: Registry,
+}
+
+impl Default for Metrics {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Metrics {
+    pub fn new() -> Self {
+        let opts = Opts::new(
+            "dcl_social_service_rpc_procedure_call",
+            "Social Service RPC Websocket Procedure Calls",
+        );
+
+        let procedure_call_collector = IntCounterVec::new(opts, &["code", "procedure"])
+            .expect("dcl_social_service_rpc_procedure_call metric can be created");
```

Comment:
Shouldn't this be the message when something fails?

Reply:
There are[ two styles](https://doc.rust-lang.org/stable/std/error/index.html#common-message-styles) for the expect messages and the recommended style is the “expect as precondition” style. I thought I was using the recommended style but clearly I'm not 🤣. I'll update it to a more a suitable one.